### ISSUE TITLE
Issue after running roslaunch

### ISSUE THREAD

**Issue by andrewraharjo** [KumarRobotics/flea3#16]
Got an issue after successfully compiling `camera_base` and `flea3` catkin_ws. FlyCap 2.9.3.43 Running Ubuntu 14.04 ROS Indigo. Any idea with what's going on ? 

```
[pg_14354944-2] process has died [pid 6529, exit code -11, cmd /home/dev-box/catkin_ws/devel/lib/flea3/flea3_single_node __name:=pg_14354944 __log:=/home/dev-box/.ros/log/d9e8303c-1e1a-11e7-95ae-ec086b69f415/pg_14354944-2.log].
log file: /home/dev-box/.ros/log/d9e8303c-1e1a-11e7-95ae-ec086b69f415/pg_14354944-2*.log

```

**Comment by versatran01** [comment]
Try upgrading to flycapture sdk 2.10, or whichever latest one from their website.

**Comment by andrewraharjo** [comment]
I don't think that would be the issue. The 2.9 running fine with `pointgrey_driver` from clearpath. When I did `rostopic list` it gives me all the publish node. Only when I `echo` them all empty

**Comment by versatran01** [comment]
Which one is giving you empty messages?

**Comment by arjun-narang** [comment]
I am getting the same error:
[pg_16493053-2] process has died [pid 9503, exit code -11, cmd /home/rj/catkin_ws/devel/lib/flea3/flea3_single_node __name:=pg_16493053 __log:=/home/rj/.ros/log/5953302a-1f3c-11e7-a9da-e4a7a0675633/pg_16493053-2.log].

Were you able to find a solution?

Note: I am using Ptgrey Chameleon3 CM3-U3-50S5C. 

Thanks!

**Comment by andrewraharjo** [comment]
@versatran01 `/pg_14354944/image_raw`
Oh by the way Chao the camera model is Grasshopper3 23S6C model.

Complete error output

```
... logging to /home/control-box/.ros/log/ee6825ea-217a-11e7-bef3-b8aeed7b71cb/roslaunch-control-vector-ai-23374.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://control-ai:37348/

SUMMARY
========

PARAMETERS
 * /pg_14354944/auto_exposure: True
 * /pg_14354944/auto_gain: True
 * /pg_14354944/auto_shutter: True
 * /pg_14354944/auto_white_balance: False
 * /pg_14354944/calib_url: file://${ROS_HOME...
 * /pg_14354944/camera_name: pg_14354944
 * /pg_14354944/exposure: True
 * /pg_14354944/fps: 7.5
 * /pg_14354944/frame_id: pg_14354944
 * /pg_14354944/gain_db: 0.0
 * /pg_14354944/height: 0
 * /pg_14354944/identifier: 14354944
 * /pg_14354944/pixel_format: 22
 * /pg_14354944/shutter_ms: 10.0
 * /pg_14354944/strobe_control: -1
 * /pg_14354944/strobe_polarity: 0
 * /pg_14354944/trigger_polarity: 0
 * /pg_14354944/trigger_source: -1
 * /pg_14354944/video_mode: 23
 * /pg_14354944/white_balance: False
 * /pg_14354944/width: 0
 * /rosdistro: indigo
 * /rosversion: 1.11.21

NODES
  /
    pg_14354944 (flea3/flea3_single_node)

auto-starting new master
process[master]: started with pid [23582]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 97b316ee-217d-11e7-ade2-dc5360c13bc6
process[rosout-1]: started with pid [23595]
started core service [/rosout]
process[pg_14354944-2]: started with pid [23604]
[ INFO] [1492220898.135311759]: /pg_14354944: Initializing reconfigure server
[pg_14354944-2] process has died [pid 23604, exit code -11, cmd /home/control-box/catkin_ws/devel/lib/flea3/flea3_single_node __name:=pg_14354944 __log:=/home/control-box/.ros/log/97b316ee-217d-11e7-ade2-dc5360c13bc6/pg_14354944-2.log].
log file: /home/control-box/.ros/log/97b316ee-217d-11e7-ade2-dc5360c13bc6/pg_14354944-2*.log

```

**Comment by andrewraharjo** [comment]
Oh by the way Chao the camera model is Grasshopper3 23S6C model.

**Comment by versatran01** [comment]
I'm still suspecting this is an issue with the flycapture sdk.

**Comment by andrewraharjo** [comment]
Chao, I'll give a shot to reinstall the Flycap SDK. Also the available SDK is 2.9 for 14.04. Are you still considering use 2.10 instead ?

**Comment by kartikmohta** [comment]
`exit code -11` usually means a segfault, can you run the node under gdb (See http://wiki.ros.org/roslaunch/Tutorials/Roslaunch%20Nodes%20in%20Valgrind%20or%20GDB) and paste a stacktrace?

**Comment by andrewraharjo** [comment]
![gdb](https://cloud.githubusercontent.com/assets/454366/25060466/e68ad2dc-2152-11e7-8306-a3e8ec95d5ca.png)

UPDATE:

So I did follow Chao suggestion on updating the FlyCap to 2.9.3.43 and still got exit code -11. 

Also my linux kernel is `4.2.0-27` Do I need to upgrade the kernel ?
Linux control 4.2.0-27-generic #32~14.04.1-Ubuntu SMP Fri Jan 22 15:32:26 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

**Comment by kartikmohta** [comment]
Can you paste the backtrace from gdb after typing `bt` in gdb after the crash.

**Comment by versatran01** [comment]
Why is it in /opt/ros/indigo/lib/libflycapture.so.2? Is it still using some lib that comes with ros?

**Comment by kartikmohta** [comment]
@versatran01 The ROS `pointgrey_camera_driver` installs it there if it's not already installed.

**Comment by versatran01** [comment]
So this could be a problem then?

**Comment by andrewraharjo** [comment]
![bt](https://cloud.githubusercontent.com/assets/454366/25065705/ec7f51a6-21c8-11e7-93b9-cb91f93cea46.png)

@kartikmohta yes you're correct the `pointgrey_camera_driver` is installed. I also have `Blackfly` connected through GigE port. Eventually we want to use both of them.

**Comment by versatran01** [comment]
So flea3 is linking to that lib that comes with ros. Maybe try force it to link to the one you installed.

**Comment by andrewraharjo** [comment]
@versatran01 Chao, where do you link that ? Is it in your `flea3/flycapture/lib/`? I think the question is more into which file does call `/opt/ros/indigo/lib/libflycapture.so.2` ?

**Comment by versatran01** [comment]
the FindFlycapture2.cmake is supposed to find the one in flea3/flycapture/lib. Just for testing, you can link directly to the one in flea3 and see if still gives an error.

**Comment by ptrouve** [comment]
Hello, I have got the same issue than andrewraharjo using FlyCap 2.9.3.43 Running Ubuntu 14.04 ROS Indigo and the camera GS3 U3 60QS6C, however I only have one installation of flycap in my computer with the lib saved in usr/lib  (no supplementary installation of flycap with ros). FindFlycapture2.cmake looks for the library in usr/lib, yet I got the exit code -11. Did direct link to the lib in flea3 solve the problem ? Do you have any other idea on this issue ?
